cmd/snap-seccomp/main.go

```diff
@@ -516,23 +516,27 @@ func addSecondaryArches(secFilter *seccomp.ScmpFilter) error {
 		// 64bit code that would have to detect at runtime if
 		// it can be used or not. But we are told this is not
 		// rare with certain classes of embedded devices.
-		compatArch = arch.UbuntuKernelArchitecture()
+		switch arch.UbuntuKernelArchitecture() {
+		case "amd64":
+			compatArch = seccomp.ArchAMD64
+		case "arm64":
+			compatArch = seccomp.ArchARM64
+		case "ppc64el":
+			compatArch = seccomp.ArchPPC64LE
+		case "ppc64":
+			compatArch = seccomp.ArchPPC64
+		case "i386":
+			compatArch = seccomp.ArchX86
+		case "armhf":
+			compatArch = seccomp.ArchARM
+		case "powerpc":
+			compatArch = seccomp.ArchPPC
+
+		}
```

Comment:
Ah yes, similar to what you do in goArchToScmpArch() in main_test.go. Perhaps since it is listed in two different places that is reason enough to factor this out now. Note that main_test.go is missing ppc64.